lib/Sema/SemaBounds.cpp

```diff
@@ -399,7 +399,11 @@ namespace {
     // should be the range of an object in memory or a subrange of
     // an object.
     BoundsExpr *LValueBounds(Expr *E) {
-      assert(E->isLValue());
+			// When type-checking errors, the resulting AST may not have 
```

Comment:
The indenting looks wrong.  If you are using Visual Studio, please be sure to set Visual Studio to insert spaces instead of tabs: Tools Options->Text Editor->All Languages->Tabs.

I think you can shorten your comment.   You don't really need to talk about assert.  It is more interesting to explain why this code is there.